spring-native/src/test/java/org/springframework/aot/test/AotCacheAwareContextLoaderDelegateTests.java

```diff
@@ -45,33 +52,107 @@
  */
 class AotCacheAwareContextLoaderDelegateTests {
 
+	private final ContextCache contextCache = mock(ContextCache.class);
+
 	private final ApplicationContext applicationContext = mock(ApplicationContext.class);
 
-	private final SmartContextLoader contextLoader = mockSmartContextLoader(this.applicationContext);
+	private final SmartContextLoader aotSmartContextLoader = mockSmartContextLoader(this.applicationContext);
+
 
 	@Test
-	void loadContextWithMatchDelegatesToSmartContextLoader() throws Exception {
-		AotCacheAwareContextLoaderDelegate delegate = createAotCacheAwareContextLoaderDelegate(
-				Map.of(SampleTest.class.getName(), () -> this.contextLoader));
-		assertThat(delegate.loadContextInternal(createMergedContextConfiguration(SampleTest.class)))
-				.isSameAs(this.applicationContext);
-		verify(this.contextLoader).loadContext(any(MergedContextConfiguration.class));
+	void loadContextWithMatchUsesAotInfrastructure() throws Exception {
+		AotCacheAwareContextLoaderDelegate delegate = createDemoDelegate();
+
+		MergedContextConfiguration mergedConfig = createMergedContextConfiguration(SampleTest.class);
+		assertThat(delegate.loadContext(mergedConfig)).isSameAs(this.applicationContext);
+
+		// loadContext() must actually be invoked with the original MergedContextConfiguration
```

Comment:
It will take me a bit of time to process this but, at a quick glance, I generally feel that a "lot' of explanations like that in a test is the sign that something else should be made more explicit.

Reply:
Fair enough.

I'll move the comments to actual Javadoc in the production code.